src/go/rpk/pkg/config/schema.go

```diff
@@ -12,14 +12,16 @@ package config
 import "path"
 
 type Config struct {
-	NodeUuid     string                 `yaml:"node_uuid,omitempty" mapstructure:"node_uuid,omitempty" json:"nodeUuid"`
-	Organization string                 `yaml:"organization,omitempty" mapstructure:"organization,omitempty" json:"organization"`
-	LicenseKey   string                 `yaml:"license_key,omitempty" mapstructure:"license_key,omitempty" json:"licenseKey"`
-	ClusterId    string                 `yaml:"cluster_id,omitempty" mapstructure:"cluster_id,omitempty" json:"clusterId"`
-	ConfigFile   string                 `yaml:"config_file" mapstructure:"config_file" json:"configFile"`
-	Redpanda     RedpandaConfig         `yaml:"redpanda" mapstructure:"redpanda" json:"redpanda"`
-	Rpk          RpkConfig              `yaml:"rpk" mapstructure:"rpk" json:"rpk"`
-	Other        map[string]interface{} `yaml:",inline" mapstructure:",remain"`
+	NodeUuid         string                 `yaml:"node_uuid,omitempty" mapstructure:"node_uuid,omitempty" json:"nodeUuid"`
+	Organization     string                 `yaml:"organization,omitempty" mapstructure:"organization,omitempty" json:"organization"`
+	LicenseKey       string                 `yaml:"license_key,omitempty" mapstructure:"license_key,omitempty" json:"licenseKey"`
+	ClusterId        string                 `yaml:"cluster_id,omitempty" mapstructure:"cluster_id,omitempty" json:"clusterId"`
+	ConfigFile       string                 `yaml:"config_file" mapstructure:"config_file" json:"configFile"`
+	Redpanda         RedpandaConfig         `yaml:"redpanda" mapstructure:"redpanda" json:"redpanda"`
+	Rpk              RpkConfig              `yaml:"rpk" mapstructure:"rpk" json:"rpk"`
+	Pandaproxy       *Pandaproxy            `yaml:"pandaproxy,omitempty" mapstructure:"pandaproxy,omitempty" json:"pandaproxy,omitempty"`
```

Comment:
should this really be pointers? Some other optional fields in Configuration uses struct and omitempty and that seems to be enough. @0x5d

Reply:
The top-level `pandaproxy` config node enables the proxy, so it has to be optional.

Reply:
sure but e.g. `KafkaApiTLS` is also optional but it's not pointer so I am wondering what's the difference :) I guess the differnce is that the TLS is enabled by Enabled field but this is enabled by the field existence ... ?

I guess then this is inconsistence on the config side, but the approach in this PR is 👍